libexec/rtld-elf/aarch64/rtld_start.S

```diff
@@ -74,7 +74,11 @@ ENTRY(.rtld_start)
 	ldr	c2, [csp, #16]			/* Load obj_main */
 	mov	c0, c19				/* Restore aux */
 	mov	csp, c20			/* Restore the stack pointer */
-	br	c8				/* Jump to the entry point */
+#if defined(__CHERI_PURE_CAPABILITY__) && defined(RTLD_SANDBOX)
+	blrr	c8				/* Jump to the entry point */
```

Comment:
Why with link?

Reply:
Because the program entry point `c8` is wrapped in a trampoline now, and the trampoline checks the Executive/Restricted mode bit of `c30` to determine the mode of the caller.

I should be able to just use `blr` here, though, since the trampoline is always Executive.

Reply:
So I can fake it by putting whatever I want (e.g. a caller's link register) in CLR and then doing a BR?..

Reply:
Yes, but I don't think this causes a security problem.

For calls that cross compartments, the callee's `c30` points back to the trampoline, so keeping that and doing a `br` has no effect, as the return portion of all trampolines are identical.

For calls that don't cross compartments, they can do whatever they want xD

I'm using `blr` here for consistency's sake. Maybe using `br` is fine, but it violates the trampoline's invariant, which is risky.

Reply:
One can also imagine putting all kinds of random stuff in `c30`, e.g., a capability to an Executive mode function (say, in `rtld`). In this case, when the callee returns, the trampoline will cause control to enter that `rtld` function, which is no different from directly calling that `rtld` function with corrupted arguments.